Mage/src/main/java/mage/ConditionalMana.java

```diff
@@ -220,4 +227,42 @@ public void add(ManaType manaType, int amount) {
                 break;
         }
     }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), conditions, staticText, scope, manaProducerId, manaProducerOriginalId);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        // Check Mana.equals(). If that's isn't equal no need to check further.
```

Comment:
Second sentence is very confusing imo due to referencing a variable named `that`.